batch7crud-roll6/src/main/java/com/lftechnology/batch7crud/controller/UserController.java

```diff
@@ -0,0 +1,214 @@
+                    logger.log(Level.SEVERE, e.getMessage());
+                }
+            } else if ("".equals(pathArgs[0]) && "edit".equals(pathArgs[2])) {
+                try {
+                    int userID = TypeCaster.toInt(pathArgs[1]);
+                    edit(request, response, userID);
+                } catch (ServletException e) {
+                    logger.log(Level.SEVERE, e.getMessage());
+                } catch (IOException e) {
+                    logger.log(Level.SEVERE, e.getMessage());
+                }
+            }
+
+        }
+
+    }
+
+    /**
+     * @see HttpServlet#doPost(HttpServletRequest request, HttpServletResponse response)
+     */
+    @Override
+    protected void doPost(HttpServletRequest request, HttpServletResponse response) throws ServletException, IOException {
+
+        String urlString = request.getPathInfo();
+        if (urlString == null) {
+            try {
+                this.list(request, response, 1);
+            } catch (ServletException e) {
+                logger.log(Level.SEVERE, e.getMessage());
+            } catch (IOException e) {
+                logger.log(Level.SEVERE, e.getMessage());
+            }
+        } else {
+            String[] pathArgs = urlString.split("/");
+            if ("".equals(pathArgs[0]) && "add".equals(pathArgs[1])) {
+                try {
+                    createProcess(request, response);
+                } catch (ServletException e) {
+                    logger.log(Level.SEVERE, e.getMessage());
+                } catch (IOException e) {
+                    logger.log(Level.SEVERE, e.getMessage());
+                }
+            } else if ("".equals(pathArgs[0]) && "edit".equals(pathArgs[2])) {
+                try {
+                    int userID = TypeCaster.toInt(pathArgs[1]);
+                    editProcess(request, response, userID);
+                } catch (ServletException e) {
+                    logger.log(Level.SEVERE, e.getMessage());
+                } catch (IOException e) {
+                    logger.log(Level.SEVERE, e.getMessage());
+                }
+            } else if ("".equals(pathArgs[0]) && "delete".equals(pathArgs[2])) {
+                try {
+                    int userID = TypeCaster.toInt(pathArgs[1]);
+                    deleteProcess(request, response, userID);
+                } catch (ServletException e) {
+                    logger.log(Level.SEVERE, e.getMessage());
+                } catch (IOException e) {
+                    logger.log(Level.SEVERE, e.getMessage());
+                }
+            }
+
+        }
+
+    }
+
+    private void list(HttpServletRequest request, HttpServletResponse response, int page) throws ServletException, IOException {
+        try {
+            request.setAttribute("users", userService.fetch(page));
+            request.getRequestDispatcher("/WEB-INF/views/listUser.jsp").forward(request, response);
+
+        } catch (DataException e) {
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png) Either log or rethrow this exception. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1166)